src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -278,14 +295,21 @@ trait ScalaSettings extends AbsScalaSettings
   def optBoxUnbox                = optEnabled(optChoices.boxUnbox)
   def optNullnessTracking        = optEnabled(optChoices.nullnessTracking)
   def optClosureInvocations      = optEnabled(optChoices.closureInvocations)
+  def optInlinerEnabled          = optEnabled(optChoices.inline)
 
-  def optInlineProject           = optEnabled(optChoices.inlineProject)
-  def optInlineGlobal            = optEnabled(optChoices.inlineGlobal)
-  def optInlinerEnabled          = optInlineProject || optInlineGlobal
+  // deprecated inliner levels
+  def optLProject                = optEnabled(optChoices.lProject)
+  def optLClasspath              = optEnabled(optChoices.lClasspath)
 
   def optBuildCallGraph          = optInlinerEnabled || optClosureInvocations
   def optAddToBytecodeRepository = optBuildCallGraph || optInlinerEnabled || optClosureInvocations
 
+  val optInlineFrom = StringSetting(
+    "-opt-inline-from",
+    "patterns",
+    "Classfile name patterns from which to allow inlining. ** = anything, * = package or class name, ! to exclude. Example: scala.**:!scala.Predef$:corp.*.util.*:corp.**.*Util*",
+    "")
```

Comment:
OK, that's a good point, I'll change it. The potentially confusing thing about a `MultiStringSetting` is that if you're using it in the last position and not in "colonated" mode, it consumes the source files. For example, this doesn't work:

```
➜  sandbox git:(inline-from) ✗ scaladoc -implicits-hide Foo Bar Test.scala
>> shows usage
➜  sandbox git:(inline-from) ✗ scaladoc -implicits-hide Foo Bar -d . Test.scala
>> runs scaladoc
```